src/main/java/org/truffleruby/core/hash/PreInitializationManager.java

```diff
@@ -40,6 +40,12 @@ public DynamicObjectFactory hookIntoHashFactory(DynamicObjectFactory originalHas
     }
 
     public void rehash() {
+        context.getRopeCache().rehash();
```

Comment:
This looks like something that would be very easy to overlook when creating a `WeakValueCache`. Can we add `PreInitializationManager` as an argument to the `WeakValueCache` constructor and force `WeakValueCache` to register itself with `PreInitializationManager`? Then this method could just iterate through the registered `WeakValueCache` instances and rehash them.

Reply:
Rehashing is due to using the seed in `Hashing`, not the WeakValueCache per se. I'll try to model this as something like `getHashing(ReHashable)`.